packages/core/solidity/src/print.ts

```diff
@@ -54,6 +56,20 @@ export function printContract(contract: Contract, opts?: Options): string {
   );
 }
 
+function printCompatibleLibraryVersions(contract: Contract): string[] {
+  const lines: string[] = [];
+  lines.push(`// Compatible with OpenZeppelin Contracts ${compatibleContractsSemver}`);
+  if (importsCommunityContracts(contract)) {
+    try {
+      const commit = getCommunityContractsGitCommit();
+      lines.push(`// Compatible with OpenZeppelin Community Contracts commit ${commit}`);
```

Comment:
I think it's worth slicing the commit to only 7 characters. Should be enough for a valid URL


```suggestion
      lines.push(`// Compatible with OpenZeppelin Community Contracts commit ${commit.slice(0, 7)}`);
```

Reply:
Truncating could lead to ambiguity in the unlikely event that multiple commits start with the same chars.

Instead, changed as follows:
- Define only 7 chars in the package.json devDependency -- if there is ambiguity at this point, yarn install gives an error.  In that case, we can use more characters (up to 40).
- Update regexes to allow the comment to contain between 7-40 chars.  GitHub URLs support that entire range.

Reply:
Yeah that makes sense. I agree it's better to have a check, and we can reuse the package.json with yarn install for that.

Thanks!